embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/payload/SessionZygote.kt

```diff
@@ -0,0 +1,43 @@
+package io.embrace.android.embracesdk.payload
+
+/**
+ * A precursor object that holds state associated with a newly started session.
+ */
+internal data class SessionZygote(
+
+    /**
+     * A unique ID which identifies the session.
+     */
+    val sessionId: String,
+
+    /**
+     * The time that the session started.
+     */
+    val startTime: Long,
+
+    /**
+     * The ordinal of the session, starting from 1.
+     */
+    val number: Int,
+
+    /**
+     * Application state for this session (foreground or background)
+     */
+    val appState: String,
+
+    /**
+     * Whether the session is a cold start or not.
+     */
+    val isColdStart: Boolean,
+
+    /**
+     * The type of start event that triggered this session.
+     */
+    val startType: Session.LifeEventType
+) {
+
+    fun toSession() = Session(
```

Comment:
Should data classes have functions? Or should Sessions have this responsibility of being initialized via a zygote?

Reply:
Or maybe just an extension function would work to keep the class completely method free.

Reply:
This is a temporary shim that I should be able to remove entirely when we no longer need a `Session` object in the payload. There are a few more fields to remove until I get to that point.

FWIW we have a detekt rule that stops arbitrary functions in data classes but makes an allowance for conversion functions like this. I felt like that was a reasonable default but happy to adopt different style if you feel strongly